app/controllers/concerns/umrdr/works_controller_behavior.rb

```diff
@@ -7,20 +7,17 @@ module WorksControllerBehavior
        
     # override curation concerns, add form fields values
     def build_form
-    
       super
-    
-      date_range = Umrdr::DateFormService.new(@form["date_coverage"].first).parse
-      if date_range
-      
-        @form["date_coverage_1_year"] = date_range[0]
-        @form["date_coverage_1_month"] = date_range[1]
-        @form["date_coverage_1_day"] = date_range[2]
-        @form["date_coverage_2_year"] = date_range[3]
-        @form["date_coverage_2_month"] = date_range[4]
-        @form["date_coverage_2_day"] = date_range[5] 
-     end  
+      setup_form_coverage
     end
+
+    # Set up the multiple parameters for the date coverage attribute in the form
+    def setup_form_coverage
```

Comment:
I see that as a potentially dangerous pattern as well.  In this case the input and recipient of the values is the same object (the `@form`).  Here we're taking the attribute (date coverage) splitting it  into its constituent parameters, and stuffing them back into a different part of the form.

This logic was originally part of the build_form method call, but some intermediate stage got a bit long.  It's reasonable to put these few lines back there.  At lest that would avoid the mysterious function.